everviz/plugins/_crossplot_plugin/crossplot_control_group_plugin.py

```diff
@@ -0,0 +1,161 @@
+                            ]
+                        ],
+                        value=None,
+                        labelStyle={"display": "inline-block"},
+                    ),
+                    "Realization 1",
+                    dcc.Dropdown(
+                        id=self.dropdown_realization_id,
+                        options=[{"label": i, "value": i} for i in dropdown_options],
+                        placeholder="Select a realization",
+                    ),
+                    "Realization 2",
+                    dcc.Dropdown(
+                        id=self.dropdown_realization_2_id,
+                        options=[{"label": i, "value": i} for i in dropdown_options],
+                        placeholder="Select a realization",
+                    ),
+                ],
+                style={"width": "60%", "display": "inline-block"},
+            )
+        ]
+
+        return html.Div(
+            basic_layout + realizations_dropdowns, style={"horizontal-align": "middle"}
+        )
+
+    def set_callbacks(self, app):
+        @app.callback(
+            Output(self.graph_id, "figure"),
+            [
+                Input(self.dropdown_x_id, "value"),
+                Input(self.dropdown_y_id, "value"),
+                Input(self.dropdown_realization_id, "value"),
+                Input(self.radio_items_x_id, "value"),
+                Input(self.radio_items_y_id, "value"),
+                Input(self.dropdown_realization_2_id, "value"),
+                Input(self.interpolation_id, "value"),
+            ],
+        )
+        def update_graph(
+            xaxis_name,
+            yaxis_name,
+            realization_nr_1,
+            xaxis_type,
+            yaxis_type,
+            realization_nr_2,
+            interpolation_type,
+        ):
+            df = get_data(self.data_path)
+            trace_1 = _get_trace(
+                df,
+                xaxis_name,
+                yaxis_name,
+                realization_nr_1,
+                xaxis_type,
+                yaxis_type,
+                interpolation_type,
+            )
+            trace_2 = _get_trace(
+                df,
+                xaxis_name,
+                yaxis_name,
+                realization_nr_2,
+                xaxis_type,
+                yaxis_type,
+                interpolation_type,
+            )
+            data = [trace_1, trace_2]
+            return _crossplot_callback(
+                data, xaxis_name, yaxis_name, xaxis_type, yaxis_type
+            )
+
+
+def _get_trace(
+    df,
+    xaxis_column_name,
+    yaxis_column_name,
+    realization_nr,
+    xaxis_type,
+    yaxis_type,
+    interpolation_type,
+):
+    try:
+        x_data = df.filter(like=xaxis_column_name).loc[realization_nr]
+        y_data = df.filter(like=yaxis_column_name).loc[realization_nr]
+    except (KeyError, TypeError):
+        return go.Scatter()
+    mode = "markers" if not interpolation_type else "markers+lines"
```

Comment:
The mode of the plot is determined based on if there is an interpolation_type specified? I think the mode should be handled higher up (eventually user defined)

Reply:
Agreed, `crossplot_control_group` will be moved to a different PR, but mode is made an argument in the current PR, so will be handled higher up.